src/components/ga4/EventBuilder/ValidateEvent/handlers/formatCheckLib.ts

```diff
@@ -0,0 +1,116 @@
+import { ValidationMessage } from "../../types"
+
+const RESERVED_EVENT_NAMES = [
+    "ad_activeview", "ad_click", "ad_exposure", "ad_impression", "ad_query",
+    "adunit_exposure", "app_clear_data", "app_install", "app_update",
+    "app_remove", "error", "first_open", "first_visit", "in_app_purchase",
+    "notification_dismiss", "notification_foreground", "notification_open",
+    "notification_receive", "os_update", "screen_view", "session_start",
+    "user_engagement"
+]
+const RESERVED_USER_PROPERTY_NAMES = [
+    "first_open_time", "first_visit_time", "last_deep_link_referrer", "user_id",
+    "first_open_after_install"
+]
+
+export const formatCheckLib = (payload) => {
+    let errors: ValidationMessage[] = []
+
+    const appInstanceIdErrors = isValidAppInstanceId(payload)
+    const eventNameErrors = isValidEventName(payload)
+    const userPropertyNameErrors = isValidUserPropertyName(payload)
+    const currencyErrors = isValidCurrencyType(payload)
+
+    return [
+        ...errors, 
+        ...appInstanceIdErrors, 
+        ...eventNameErrors,
+        ...userPropertyNameErrors,
+        ...currencyErrors,
+    ]
+}
+
+const isValidAppInstanceId = (payload) => {
+    let errors: ValidationMessage[] = []
+    const appInstanceId = payload.app_instance_id
+
+    if (appInstanceId.length !== 32) {
+        errors.push({
+            description: `Measurement app_instance_id is expected to be a 32 digit hexadecimal number but was [${appInstanceId.length}] digits.`,
+            validationCode: "FormatCheckError",
+            fieldPath: "app_instance_id"
+        })
+    }
+
+    if (!appInstanceId.match(/[a-f0-9]/)) {
+        let nonChars = appInstanceId.split('').filter(letter => {
```

Comment:
Can you please fix the warning "Array.prototype.filter() expects a value to be returned at the end of arrow function" ?

Reply:
Changed to forEach

Reply:
This filter is actually returning letter on line 48. Should I disable the warning? Or am I missing a better way to do this?